pkg/filesystem/subvolume.go

```diff
@@ -166,36 +205,20 @@ func unMarshaljson(list string) []fsStruct {
 	return unmarshal
 }
 
-func Delete(ctx context.Context, clientsets *k8sutil.Clientsets, OperatorNamespace, CephClusterNamespace, subList, fs, svg string) {
-	subvollist := strings.Split(subList, ",")
+func Delete(ctx context.Context, clientsets *k8sutil.Clientsets, OperatorNamespace, CephClusterNamespace, fs, subvol, svg string) {
 	k8sSubvolume := getK8sRefSubvolume(ctx, clientsets)
-	for _, subvolume := range subvollist {
-		check := checkStaleSubvolume(ctx, clientsets, OperatorNamespace, CephClusterNamespace, fs, subvolume, svg, k8sSubvolume)
-		if check {
-			_, err := exec.RunCommandInOperatorPod(ctx, clientsets, "ceph", []string{"fs", "subvolume", "rm", fs, subvolume, svg}, OperatorNamespace, CephClusterNamespace, true)
-			if err != nil {
-				logging.Error(err, "failed to delete stale subvolume %q", subvolume)
-				continue
-			}
-			logging.Info("subvolume %q deleted", subvolume)
-		} else {
-			logging.Info("subvolume %q is not stale", subvolume)
-		}
+	check := isStaleSubvolume(subvol, k8sSubvolume)
+	if !check {
+		exec.RunCommandInOperatorPod(ctx, clientsets, "ceph", []string{"fs", "subvolume", "rm", fs, subvol, svg, "--retain-snapshots"}, OperatorNamespace, CephClusterNamespace, true)
+		logging.Info("subvolume %q deleted", subvol)
+	} else {
+		logging.Info("subvolume %q is not stale", subvol)
 	}
 }
 
-// checkStaleSubvolume checks if there are any stale subvolume to be deleted
-func checkStaleSubvolume(ctx context.Context, clientsets *k8sutil.Clientsets, OperatorNamespace, CephClusterNamespace, fs, subvolume, svg string, k8sSubvolume map[string]subVolumeInfo) bool {
+// isStaleSubvolume checks if there are any stale subvolume to be deleted
+func isStaleSubvolume(subvolume string, k8sSubvolume map[string]subVolumeInfo) bool {
```

Comment:
```suggestion
func isStaleSubvolume(subvolume string, k8sSubvolume map[string]subVolumeInfo) bool {
```

do we need this function? I see this is a one-liner method and only called once. So, will suggested to not include in new function